xmlconf/src/xmlconf.zig

```diff
@@ -202,25 +211,29 @@ fn runTest(gpa: Allocator, dir: std.fs.Dir, reader: *xml.Reader, results: *Resul
     const input = input: {
         const index = reader.attributeIndex("URI") orelse return error.InvalidTest;
         const path = try reader.attributeValue(index);
-        break :input dir.readFileAlloc(gpa, path, max_file_size) catch |err|
+        const buf = try gpa.alloc(u8, max_file_size);
+        const input = dir.readFile(std.testing.io, path, buf) catch |err|
             return results.err("{s}: reading input file: {s}: {}", .{ id, path, err });
+        break :input .{ .parent = buf, .data = input };
     };
-    defer gpa.free(input);
-    const output = output: {
-        const index = reader.attributeIndex("OUTPUT") orelse break :output null;
+    defer gpa.free(input.parent);
+    const output: struct { parent: ?[]u8, data: ?[]u8 } = output: {
```

Comment:
this feels v janky but was having an invalid free because (I think?) the readFile returned a view into the greater buf slice and then when we freed that we freed only the view not the full slice. im sure theres a better way but this works/tests ran/no allocator complaints

Reply:
It looks like there's a TODO to move `readFileAlloc` to `std.Io.Dir`: https://github.com/ziglang/zig/blob/bdbfc7de3fd3a05f338ec8beaa29894547a9df4c/lib/std/fs/Dir.zig#L1425-L1426 So, hopefully, this workaround will be unnecessary in the long run. (If it doesn't get moved for some reason, I would probably end up just copying the core logic of `readFileAlloc`)

Reply:
copy that, will keep an eye and update if/when stuff changes or do as you suggested!